brave/src/test/java/brave/handler/MutableSpanTest.java

```diff
@@ -578,7 +579,7 @@
         MutableSpan other = otherConstructor.get();
         assertThat(span).isNotSameAs(other); // sanity
         assertNeitherEqualNorShareHashCode(span, other);
-        assertThat(span).usingRecursiveComparison().isNotEqualTo(other); // double check our impl
+        assertThat(Objects.deepEquals(span, other)).isFalse(); // double check our impl
```

Comment:
assertj's recursive can't tell the difference between two exceptions with the same value state anymore